### PR TITLE
KUBESAW-43: Replace the special ToolchainClusterCondition with the standard toolchain Condition

### PR DESCRIPTION
**Pull request by fbm3307**
This is to replace the special ToolchainClusterCondition with the standard toolchain Condition.

This is Related to Change in

- API - https://github.com/codeready-toolchain/api/pull/416
- Toolchain-Common - https://github.com/codeready-toolchain/toolchain-common/pull/391
- Host-Operator - https://github.com/codeready-toolchain/host-operator/pull/1017
- Member-Operator - https://github.com/codeready-toolchain/member-operator/pull/560
- Toolchain-E2E - https://github.com/codeready-toolchain/toolchain-e2e/pull/955

Only change is updating the dependency of API and toolchain-common

- Registration-Service - https://github.com/codeready-toolchain/registration-service/pull/424
- Sandbox-SRE - https://github.com/codeready-toolchain/sandbox-sre/pull/1706

### CODE REVIEW

**Review by mfrancisc — APPROVED** [review]

File: pkg/cmd/adm/register_member.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func waitUntilToolchainClusterReady(ctx *clicontext.CommandContext, cl runtimecl
 
 		// TODO: replace with condition.IsTrue() once the ToolchainClusterCondition is replaced by the generic one.
 		for _, cond := range tc.Status.Conditions {
-			if cond.Type == toolchainv1alpha1.ToolchainClusterReady && cond.Status == corev1.ConditionTrue {
+			if cond.Type == toolchainv1alpha1.ConditionReady && cond.Status == corev1.ConditionTrue {
 				return true, nil
 			}
 		}
```
[the remark this answers is not in the document]
Comment (metlos) [reply pkg/cmd/adm/register_member.go:198]: Please don't commit the suggestion directly though. It wouldn't compile... Github didn't allow me to also remove the line following the removed lines (ie. the `return false, nil`).

[the remark this answers is not in the document]
Comment (fbm3307) [reply pkg/cmd/adm/register_member.go:198]: Hey @metlos Thanks, i missed the comment in above code, updated it now

**Comment by fbm3307** [comment]
@metlos @MatousJobanek can you please provide an review/approval to this PR ?

**Review by MatousJobanek — COMMENTED** [review]
Looks good, sorry for the delayed review

**Pull request merged**